### PR TITLE
Added digging handler

### PR DESCRIPTION
**Pull request by zhuowei**
I've added a digging handler to respond to digging messages. It now allows players to break blocks.

### CODE REVIEW

**Comment by grahamedgecombe** [comment]
Thanks for the patch. I've pulled it and did another commit so the conventions match what I use, and I also added a few TODO considerations.

**Comment by davorb** [comment]
A little "feature" that you might want to look into eventually... if you dig under water it replaces the block with air instead of water.

File: src/net/lightstone/msg/handler/DiggingMessageHandler.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	public void handle(Session session, Player player, DiggingMessage message) {
+		if(message.getState()==2){ //done digging
+			int x=message.getX();
+			int z=message.getZ();
+			int chunkX=x/Chunk.WIDTH+((x<0&&x%16!=0)?-1:0);
+			int chunkZ=z/Chunk.HEIGHT+((z<0&&z%16!=0)?-1:0);
+			Chunk chunk=session.getServer().getWorld().getChunks().getChunk(chunkX, chunkZ);
+			int localX=(x-(chunk.getX()*Chunk.WIDTH))%Chunk.WIDTH;
+			int localZ=(z-(chunk.getZ()*Chunk.HEIGHT))%Chunk.HEIGHT;
+			chunk.setType(localX, localZ, message.getY(), 0); //set to air
```
Comment (davorb) [inline src/net/lightstone/msg/handler/DiggingMessageHandler.java:25]: Need to add checks for if the player is under water here

**Comment by davorb** [comment]
We could let the physics/flow thingie handle it as well. But it's most likely less resource intensive if we do the physics checks only when people actually change stuff.

**Pull request merged**